src/daft-local-execution/src/intermediate_ops/intermediate_op.rs

```diff
@@ -28,11 +35,11 @@ pub trait IntermediateOperator: Send + Sync {
         &self,
         idx: usize,
         input: &PipelineResultType,
-        state: Option<&mut Box<dyn IntermediateOperatorState>>,
+        state: &mut dyn IntermediateOperatorState,
```

Comment:
Didn't like the `Option<&mut Box<dyn IntermediateOperatorState>>` parameter, instead changing it to just `&mut dyn IntermediateOperatorState`

Reply:
agreed, this is cleaner